### ISSUE TITLE
Tempo ignores Zipkin error tag

### ISSUE THREAD

**Issue by jonatan-ivanov** [grafana/tempo#1916]
**Describe the bug**
The standard way of indicating that an operation/span has failed in Zipkin is the presence of the `error` tag. The value of the `error` tag is "the message or simple name of the throwable" (see the [documentation](https://javadoc.io/doc/io.zipkin.brave/brave/latest/brave/Tags.html#ERROR))
It seems Tempo ignores this and none of the spans are marked as failed if the value of the `error` tag is not `true`. This is wrong since the value of the `error` tag should be the error message and `true` is a weird error message (also not very useful). This also makes finding the root cause in big traces hard.

**To Reproduce**
1. Start Tempo (`latest` docker image ID of `grafana/tempo`: `7ab6a4d0b970`)
2. Send a span in Zipkin format that contains the `error` tag with an error message (e.g.: `error=Ooops!`)
3. Look at the span in Tempo, the error marker does not show up

**Expected behavior**
Error spans even if they are sent in the Zipkin format should have the error marker in Tempo.

**Environment:**
 - Infrastructure: local Docker
 - Deployment tool: Docker

**Additional Context**
Here's an example span in Zipkin format that you can send to Tempo:

```json
[
    {
        "traceId": "4836213d48967e81145cd7ebbefe2292",
        "id": "ca7d12f0985eff8d",
        "kind": "SERVER",
        "name": "http get /trouble",
        "timestamp": 1669688256135850,
        "duration": 64240,
        "localEndpoint": {
            "serviceName": "web-sample",
            "ipv4": "192.168.0.101"
        },
        "tags": {
            "error": "Noooooo!"
        }
    }
]
```

**Comment by joe-elliott** [comment]
Thanks for the report. We use opentelemetry's translators when converting from Jaeger and Zipkin to OTLP:

https://github.com/open-telemetry/opentelemetry-collector-contrib/tree/main/pkg/translator/zipkin

Digging in the code it does seem hard coded to only respect the string "true" for setting the error status:

https://github.com/open-telemetry/opentelemetry-collector-contrib/blob/97eb6ada98a0083ec12080ddddd68ac09500259f/pkg/translator/zipkin/zipkinv2/to_translator.go#L168-L169

I've filed an issue here:

https://github.com/open-telemetry/opentelemetry-collector-contrib/issues/16530

Please add any additional information to the OTEL issue linked above. You have far more expertise here than myself! This fix seems simple enough so hopefully we can something merged into OTEL and then revendor it into Tempo.

**Comment by jonatan-ivanov** [comment]
Thank you for the answer and opening the issue.
In this case it seems the OTel collector does not follow the OTel Zipkin exporter specification: https://opentelemetry.io/docs/reference/specification/trace/sdk_exporters/zipkin/#status

**Comment by jonatan-ivanov** [comment]
Hi,

Could somebody please label the issue appropriately so that it is not going to stale?
I guess the issue is blocked on https://github.com/open-telemetry/opentelemetry-collector-contrib/issues/16530 but I think it is still valid.

**Comment by joe-elliott** [comment]
I'm looking at the PR where they fixed the issue:

https://github.com/open-telemetry/opentelemetry-collector-contrib/pull/24547/files

I only see this change:

https://github.com/open-telemetry/opentelemetry-collector-contrib/pull/24547/files#diff-ffbd1966d5b60729b7588dec99ea906188b8378195af4a8aa6b263b95cf311bfR158

which looks like it sets the error status for any value of the `error` tag instead of only setting it on `true`. You can see on this line:

https://github.com/open-telemetry/opentelemetry-collector-contrib/pull/24547/files#diff-ffbd1966d5b60729b7588dec99ea906188b8378195af4a8aa6b263b95cf311bfR160

where it removes the error tag but only if it's equal to true. So it seems like we always should have been storing the error tag/value? We just weren't setting the span status appropriately?

**Comment by joe-elliott** [comment]
This is an OTel issue.
